server/repository/src/db_diesel/store_tag_row.rs

```diff
@@ -0,0 +1,248 @@
+                continue;
+            }
+
+            let store_tag_row = StoreTagRow {
+                id: format!("{}_{}", store_id.clone(), tag_name),
+                store_id: store_id.clone(),
+                tag_name: tag_name.to_string(),
+            };
+            self.insert_one(&store_tag_row)?;
+        }
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod test_store_tag_row {
+    use crate::{
+        test_db, NameRow, NameRowRepository, StoreRow, StoreRowRepository, StoreTagRow,
+        StoreTagRowRepository, StoreTags,
+    };
+
+    fn tags_match(expected: Vec<String>, actual: Vec<StoreTagRow>) -> Result<bool, String> {
+        let tag_names: Vec<String> = actual.iter().map(|tag| tag.tag_name.clone()).collect();
+        if expected != tag_names {
+            return Err(format!(
+                "Expected tags: {:?}, Actual tags: {:?}",
+                expected, tag_names
+            ));
+        }
+        Ok(true)
+    }
+
+    #[actix_rt::test]
+    async fn test_store_tag_repository() {
+        let settings = test_db::get_test_db_settings("omsupply-database-test_store_tag_repository");
+        let connection_manager = test_db::setup(&settings).await;
+        let connection = connection_manager.connection().unwrap();
+
+        // setup
+        NameRowRepository::new(&connection)
+            .insert_one(&NameRow {
+                id: "name1".to_string(),
+                name: "name1".to_string(),
+                code: "name1".to_string(),
+                ..Default::default()
+            })
+            .await
+            .unwrap();
+
+        StoreRowRepository::new(&connection)
+            .insert_one(&StoreRow {
+                id: "store1".to_string(),
+                name_id: "name1".to_string(),
+                ..Default::default()
+            })
+            .await
+            .unwrap();
+
+        /* TESTS */
+
+        let repo = StoreTagRowRepository::new(&connection);
+
+        // Check we can insert a store tag
+        let store_tag_row = StoreTagRow {
+            id: "store1_tag1".to_string(),
+            store_id: "store1".to_string(),
+            tag_name: "tag1".to_string(),
+        };
+
+        repo.insert_one(&store_tag_row).unwrap();
+
+        // Check we can find a store tag by id
+        let found_store_tag = StoreTagRowRepository::new(&connection)
+            .find_one_by_id(&store_tag_row.id)
+            .unwrap();
+        assert_eq!(found_store_tag, Some(store_tag_row));
+
+        // Check that the store tag list is upserted correctly with the same list of tags
+        let store_tags = StoreTags {
+            store_id: "store1".to_string(),
+            tags: vec!["tag1".to_string()],
+        };
+        repo.upsert_tags_for_store(&store_tags).unwrap();
+
+        let current_tags = repo.find_all_by_store_id(&store_tags.store_id).unwrap();
+        assert!(tags_match(store_tags.tags, current_tags).unwrap());
```

Comment:
hmm this is a little bit weird use of assert, since result is always true, personally I would have gone with:
```rust
assert_eq!(store_tags.tags, current_tags.into_iter().map(|t| t.tag_name).collect::<Vec<String>>());
```